### PR TITLE
cargo-valgrind: init at 1.3.0

### PR DESCRIPTION
**Pull request by otavio**
###### Motivation for this change

Introduce a new useful tool for Rust users

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by otavio** [comment]
Result of `nixpkgs-review pr 108388` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-valgrind</li>
  </ul>
</details>

File: pkgs/development/tools/rust/cargo-valgrind/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "jfrimmel";
+    repo = "cargo-valgrind";
+
+    # repo lacks the 1.3.0 tag
+    # https://github.com/jfrimmel/cargo-valgrind/issues/33
+    rev = "d47dd810e3971d676cde5757df8b2e05ed563e41";
+
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-valgrind/default.nix:18]:
```suggestion
    # repo lacks the 1.3.0 tag
    # https://github.com/jfrimmel/cargo-valgrind/issues/33
    rev = "d47dd810e3971d676cde5757df8b2e05ed563e41";
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108388` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cargo-valgrind</li>
</ul>
</details>

**Comment by otavio** [comment]
@SuperSandro2000 I forced-push squashing your proposed change. Please take a look at it.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108388` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cargo-valgrind</li>
</ul>
</details>

**Review by Ma27 — CHANGES_REQUESTED** [review]
You should probably create a wrapper here to make sure that `valgrind` can be found even if it's not installed globally or in a local profile.

**Comment by otavio** [comment]
Oh, good. I'll research how to do it.

**Comment by otavio** [comment]
Oh, good. I'll research how to do it.

> You should probably create a wrapper here to make sure that `valgrind` can be found even if it's not installed globally or in a local profile.

I'll look very dumb but I did not find out how to do this. Do you have an example or documentation I could take look at?

**Comment by Ma27** [comment]
you basically want to use [makeWrapper](https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-functions) for that. The bash-code in a post-install hook may look like this then:

``` nix
/* ... */ {
  nativeBuildInputs = [ makeWrapper ];
  postInstall = ''
    wrapProgram $out/bin/cargo-valgrind --prefix PATH : ${stdenv.lib.makeBinPath [ valgrind ]}
  '';
}
```

**Comment by otavio** [comment]
It kinda work. The `valgrind` is not installed ... dunno why.

```
error: valgrind executable not found in `PATH` (No such file or directory (os error 2))
```

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 108388 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>cargo-valgrind</li>  </ul></details>

**Pull request merged**